Sources/StreamChat/Utils/EventBatcher_Tests.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChat
+import XCTest
+
+final class Batch_Tests: StressTestCase {
+    func test_append() {
+        // Create batcher for test events and keep track of handler calls
+        var handlerCalls = [[TestEvent]]()
+        let batcher = Batcher<TestEvent>(period: 0.2) { events in
+            handlerCalls.append(events)
+        }
+
+        // Prepare some batches of events
+        let event1 = TestEvent()
+        let event2 = TestEvent()
+        
+        // Add 1st event to batch
+        batcher.append(event1)
+        
+        // Wait a bit less then period
+        wait(0.05)
+        // Assert current batch contains expected values
+        XCTAssertEqual(batcher.currentBatch, [event1])
+        // Assert handler is not called yet
+        XCTAssertEqual(handlerCalls, [])
+        
+        // Add 2nd event to batch
+        batcher.append(event2)
+        
+        // Wait a bit less then period
+        wait(0.05)
+        // Assert current batch contains expected values
+        XCTAssertEqual(batcher.currentBatch, [event1, event2])
+        // Assert handler is not called yet
+        XCTAssertEqual(handlerCalls, [])
+        
+        // Wait another bit so batch period has passed
+        wait(0.2)
```

Comment:
So many `wait`s in these tests makes me thing the component is not very testable.. How about if we inject a Timer and use it instead of `queue.asyncAfter`?

Reply:
Yep, that was my initial idea 👍 If we inject time we still need a serial queue to prevent issues with appending elements in a multithreaded environment. And with both queue + time + timer the implementation gets messy in favor of tests. Multiple `wait` statements can lead to test timeout but we haven't seen these issues for quite some time and if we will that would be a matter of decreasing the batch period used.

Reply:
@evsaev You can make sure a Timer fires it's blocks in a given queue, check DispatchSourceTimer here: https://developer.apple.com/documentation/dispatch/dispatchsource/2300106-maketimersource